dashboard/api.py

```diff
@@ -684,10 +689,20 @@ def calculate_users_to_refresh_in_bulk():
         usercacherefreshtime__current_grade__gte=refresh_time_limit
     )
 
+    only_dedp_users = []
+    if not settings.UPDATE_EDX_DATA_FOR_DEDP_PROGRAM_USERS:
+        dedp_users = ProgramEnrollment.objects.filter(
+            program__title=DEDP
+        ).values_list('user__id', flat=True)
+        non_dedp_users = ProgramEnrollment.objects.filter(
+            Q(user__id__in=dedp_users), ~Q(program__title=DEDP)
```

Comment:
could you add a test for this?

Reply:
you could also do .exclude(program__title=DEDP).filter(user_id__in=dedp_users) which I think is more readable